src/bin/wasm-tools/objdump.rs

```diff
@@ -36,15 +38,27 @@ impl Opts {
                 CodeSectionStart { range, count, .. } => printer.section_raw(range, count, "code"),
                 CodeSectionEntry(_) => {}
 
-                ComponentTypeSection(_) => todo!("component-model"),
-                ComponentImportSection(_) => todo!("component-model"),
-                ComponentFunctionSection(_) => todo!("component-model"),
-                ModuleSection { .. } => todo!("component-model"),
-                ComponentSection { .. } => todo!("component-model"),
-                InstanceSection(_) => todo!("component-model"),
-                ComponentExportSection(_) => todo!("component-model"),
-                ComponentStartSection { .. } => todo!("component-model"),
-                AliasSection(_) => todo!("component-model"),
+                ComponentTypeSection(s) => printer.section(s, "types"),
+                ComponentImportSection(s) => printer.section(s, "imports"),
+                ComponentFunctionSection(s) => printer.section(s, "functions"),
+                ModuleSection { range, .. } => { 
+                    printer.section_raw(range, 1, "module");
+                    if let Some(space) = printer.indices.last_mut() {
+                        space.inc_module();
+                    }
+                },
+
+                ComponentSection { range, .. } => {
+                    let mut index_space = IndexSpace::default();
+                    index_space.inc_component();
```

Comment:
Should this be incremented on the parent index space similar to how it's performed for `ModuleSection`?

Reply:
Good catch, yeah! I applied your suggestion to simplify the increment logic and this is now done when ending a component.